src/Objs/SearchResults/SearchResultsComponent.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  currentPage,
+  ImageTag,
+  InPlaceEditingOff,
+  load,
+  navigateTo,
+  Obj,
+  provideComponent,
+} from 'scrivito'
+import { useEffect, useRef, useState } from 'react'
+import { ensureString } from '../../utils/ensureString'
+import { SearchResult, SearchResultLoadingPlaceholder } from './SearchResult'
+import { SearchResults } from './SearchResultsObjClass'
+import { DATA_OBJ_CLASSES } from '../dataObjClasses'
+
+const BLACKLIST_OBJ_CLASSES = [
+  'Image',
+  'SearchResults',
+  'Video',
+  ...DATA_OBJ_CLASSES,
+]
+
+provideComponent(SearchResults, ({ page, params }) => {
+  const query = ensureString(params?.q).trim()
+
+  const inputRef = useRef<HTMLInputElement>(null)
+  const [{ maxItems, searchResults, totalCount, updateCounter }, setState] =
+    useState<{
+      maxItems: number
+      searchResults: Array<Obj> | null
+      totalCount: number | null
+      updateCounter: number
+    }>({
+      maxItems: 10,
+      searchResults: null,
+      totalCount: null,
+      updateCounter: 0,
+    })
+
+  useEffect(() => {
+    const search = Obj.whereFullTextOf('*', 'matches', query)
+      .andNot('_objClass', 'equals', BLACKLIST_OBJ_CLASSES)
+      .and('_dataParam', 'equals', null) // Ignore data details pages
+      .andNot('excludeFromSearch', 'equals', true)
+
+    load(() => [search.take(maxItems), search.count()] as const).then(
+      ([searchResults, totalCount]) => {
+        setState((prevState) => {
+          if (prevState.updateCounter !== updateCounter) return prevState
+
+          return {
+            ...prevState,
+            searchResults,
+            totalCount,
+            updateCounter: prevState.updateCounter + 1,
+          }
+        })
+      },
+    )
+  }, [query, maxItems, updateCounter])
```

Comment:
FYI: This currently produces an infinite rerender loop (because a changed `updateCounter` also tiggers a new load). I'll look into this.